test/unit/double_free.c

```diff
@@ -101,10 +101,44 @@ TEST_BEGIN(test_small_double_free_tcache) {
 }
 TEST_END
 
+TEST_BEGIN(test_small_double_free_arena) {
+	test_skip_if(!config_debug);
+
+	test_skip_if(opt_debug_double_free_max_scan == 0);
+
+	bool tcache_enabled;
+	size_t sz = sizeof(tcache_enabled);
+	assert_d_eq(
+	    mallctl("thread.tcache.enabled", &tcache_enabled, &sz, NULL, 0), 0,
+	    "Unexpected mallctl failure");
+	test_skip_if(!tcache_enabled);
```

Comment:
nit: now that this section is used twice in this file, make it a separate function.

Reply:
Added a function to club the tcache enabling. Yet test_skip_if cannot be called outside the test so still keep the lines 113 and 115

Reply:
nit: Let's do `test_skip_if(tcache_enabled());` (rename the function because enable_tcache() implies it's doing something instead of checking a state). Also probably remove the blank lines between the 3 test_skips here.